tests/util/pool_tests.rs

```diff
@@ -61,6 +68,78 @@ pub fn test_thinpool_expand(paths: &[&Path]) -> () {
 }
 
 
+/// Verify a snapshot has the same files and same contents as the origin.
+pub fn test_filesystem_snapshot(paths: &[&Path]) {
+    let dm = DM::new().unwrap();
+    let (mut pool, _) =
+        StratPool::initialize("stratis_test_pool", &dm, paths, Redundancy::NONE, true).unwrap();
+    let &(_, fs_uuid) = pool.create_filesystems(&[("stratis_test_filesystem", None)])
+        .unwrap()
+        .first()
+        .unwrap();
+    let write_buf = &[8u8; SECTOR_SIZE];
+    let file_count = 10;
+    let source_tmp_dir = TempDir::new("stratis_testing").unwrap();
+    {
+        // to allow mutable borrow of pool
+        let filesystem = pool.get_mut_strat_filesystem(&fs_uuid).unwrap();
+        println!("Mount {:?} on {:?}",
+                 filesystem.devnode(),
+                 source_tmp_dir.path());
+        mount(Some(&filesystem.devnode()),
+              source_tmp_dir.path(),
+              Some("xfs"),
+              MS_VERBOSE,
+              None as Option<&str>)
+                .unwrap();
+        for i in 0..file_count {
+            let file_path = source_tmp_dir
+                .path()
+                .join(format!("stratis_test{}.txt", i));
+            let mut f = OpenOptions::new()
+                .create(true)
+                .write(true)
+                .open(file_path)
+                .unwrap();
+            f.write_all(write_buf).expect("write to file");
+            f.flush().expect("flush snapshot source");
+
+        }
+
+    }
+    // Run a check to expand the pool.  This is necessary because
+    // the snapshot uses all the same blocks in the pool until we
+    // update the UUID.  That causes some of the blocks to
+    // consume space in the pool.
+    pool.check();
```

Comment:
Make sure to rebase on current master, this will need to be an unwrap().

Reply:
done